packages/eslint-config/src/index.mjs

```diff
@@ -13,6 +13,78 @@ import globals from 'globals';
 import { ignores } from "./ignores.mjs";
 import { rules } from "./rules.mjs";
 
+import path from 'path';
+import { fileURLToPath } from 'url';
+
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = path.dirname(__filename);
+const projectRoot = path.resolve(__dirname, '../../..');
+
+const packages = [
```

Comment:
This is a major part of the fix - per-project linting definitions so that eslint uses the appropriate rules, and the correct tsconfig is used for each package for linting purposes.